File: LeetCode/C#/543. Diameter of Binary Tree.cs
using System;
public class Solution {
    public int DiameterOfBinaryTree(TreeNode root) 
    {
         int max = 0;

        Depth(root);

        return max;

        int Depth(TreeNode node)
        {
            if (node == null)
            {
                return 0;
            }

            int left = Depth(node.left);
            int right = Depth(node.right);
            max = Math.Max(max, left + right);

            return Math.Max(left, right) + 1;
        }
    }
}

File: LeetCode/Python/543. Diameter of Binary Tree.py
# type: ignore
class Solution:
    def diameterOfBinaryTree(self, root: Optional[TreeNode]) -> int:
        
        def depth(root):
            if not root:
                return 0
            left = depth(root.left)
            right = depth(root.right)
            self.ans = max(self.ans, left + right)
            return max(left, right) + 1
        
        self.ans = 0
        depth(root)
        return self.ans